src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const copy = [...this];
+    let result = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (copy[copy.length - 1] === undefined) {
+      copy[copy.length - 1] = '';
+    }
+
+    for (let i = 0; i < copy.length - 1; i++) {
+      if (copy[i] === null || copy[i] === undefined) {
+        copy[i] = '';
+      }
```

Comment:
let's add the item to the result string only if it's not empty or undefined instead of setting empty string to the element